exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java

```diff
@@ -157,6 +157,967 @@ public void testRegexpReplace() throws Exception {
   }
 
   @Test
+  public void testLikeStartsWith() throws Exception {
+
+    // all ASCII.
+    testBuilder()
```

Comment:
The regex parsing and execution code is becoming complex. Let's test it with a true unit test, not just a system-level test using a query. See the test frameworks available. We can also discuss in person.